### PR TITLE
Implement ListDownloadCertificate Command

### PR DESCRIPTION
**Pull request by DechengMa**
This PR is to Implement the list and download certificates feature.
Closes issue #86 

# 📝 Summary of Changes

Changes proposed in this pull request:

- Add `serialNumber` to model `Certificate` 
- Create `ListCertificatesOptions` and `ListCertificatesOpertaion`
- Implement `ListDownloadCertificates`, support viewing and downloading certificates

# 🧐🗒 Reviewer Notes

## 💁 Example

Usage: `asc certificates list [--filter-serial <filter-serial>] [--sort <sort>] [--filter-type <filter-type>] [--filter-display-name <filter-display-name>] [--limit <limit>] [--download-path <download-path>]`

Sample output:
- without download file path: 

| SerialNumber | Name | Display Name |  Platform | Type | Expiration Date | 
| --- | --- | --- | --- | --- | --- |
| XXXYY | iOS Development: Created via API | Something | IOS | IOS_DEVELOPMENT | 2021-04-22 01:23:45 +0000 | 

- with download file path:

`📥 Certificate 'iOS Development: Created via API' downloaded to: /Documents/github/XXXYY.cer`
| SerialNumber | Name | Display Name |  Platform | Type | Expiration Date | 
| --- | --- | --- | --- | --- | --- |
| XXXYY | iOS Development: Created via API | Something | IOS | IOS_DEVELOPMENT | 2021-04-22 01:23:45 +0000 | 

## 🔨 How To Test
`swift run asc certificates list`

`swift run asc certificates list --download-path ~/Documents/github `

### CODE REVIEW

File: Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        if let downloadPath = downloadPath {
+            try certificates.forEach { (certificate: Certificate) in
+                guard let content = certificate.content else {
+                    throw CommandError.invalidContent
+                }
+
+                let filePath = "\(downloadPath)/\(certificate.serialNumber ?? "serial").cer"
+
+                guard FileManager
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift:77]: We're using the `Files` swift module to help with file operations.  Check https://github.com/johnsundell/files

**Comment by orj** [comment]
This is looking good @DechengMa, just some minor changes would make it 💯

File: Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            try certificates.forEach { (certificate: Certificate) in
+                guard let content = certificate.content else {
+                    throw CommandError.invalidContent
+                }
+
+                do {
+                    let folder = try Folder(path: downloadPath)
+
+                    let file = try folder.createFile(
+                        named: "\(certificate.serialNumber ?? "serial").cer",
```
Comment (adamjcampbell) [inline Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift:80]: Is this the expected default. Or should we throw an error if we don't get a name?

Comment (adamjcampbell) [reply Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift:80]: Or serial sorry

Comment (DechengMa) [reply Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift:80]: the most important thing about the certificate is its content. I don't quite sure if a certificate can only have `content`, without a `serialNumber `. What do you think?

Comment (adamjcampbell) [reply Sources/AppStoreConnectCLI/Commands/Certificates/ListDownloadCertificates.swift:80]: I'm not sure, let's progress as is but we should think about declaring what we think is non optional and throw errors when we don't get those things in the future

**Pull request merged**